### PR TITLE
I/7579: Allow disabling image resize handles 

### LINKED ISSUE #7579: Allow disabling resize handles when using resize buttons
## Provide a description of the task

The `ImageResize` plugin was refactored to common glue plugin that requires:

*   `ImageResizeEditing` - for command **and** image resize handles
*    `ImageResizeUI` - for dropdown and buttons

However, we've identified that not everybody would like to have buttons and resize handles and it would be nice to configure `ImageResizeEditing` feature so that handles are optional (as opt-out to have backward compatibility if possible).

I can see two options:

1.  Split the `ImageResizeEditing` to two - but the question remains how to name the resulting plugins (and how to keep backward compatibility for previous `ImageResize` plugin - or introduce breaking change.
2.  Use configuration option, ie: `config.imageResizeHandles: false` (or similar).

### PR DESCRIPTION
**Pull request by panr**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Other (image): Allow to configure `ImageResize` in a more granular way. For example, by combining `ImageResizeEditing` with `ImageResizeHandles` or `ImageResizeUI` to resize image by handles or by image toolbar UI components (dropdown or standalone buttons) respectively. Closes #7579.

Docs (image): Add a new section to feature documentation for the `Image` plugin about disabling image resize handles.

Internal (image): Separate `ImageResizeHandles` plugin from the `ImageResizeEditing`. 

Tests (image): Refactor unit tests for `ImageResizeHandles` and `ImageResizeEditing`.

### CODE REVIEW

**Comment by panr** [comment]
I don't get it. Locally everything is 🟢 🤔 (all tests and `yarn docs --strict`).

**Comment by jodator** [comment]
> I don't get it. Locally everything is green\_circle thinking (all tests and `yarn docs --strict`).

There was dev-tools update AFAIR and Kamil might be working on it.

File: packages/ckeditor5-image/docs/features/image.md [hunk 1]
```diff
@@ -322,7 +444,7 @@ ClassicEditor
 
 ## Linking images
 
-The {@link module:link/linkimage~LinkImage} plugin adds support for linking images. Some use cases where this is needed are:
+The {@link module:link/linkimage, ... adds support for linking images. Some use cases where this is needed are:
```
Comment (jodator) [inline packages/ckeditor5-image/docs/features/image.md:447]: Something broken here.

File: packages/ckeditor5-image/src/imageresize.js [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
 import ImageResizeUI from './imageresize/imageresizeui';
```
Comment (jodator) [inline packages/ckeditor5-image/src/imageresize.js:11]: Let's have it as `ImageResizeButtons`. (ref: https://github.com/ckeditor/ckeditor5/issues/7579#issuecomment-655487065)

File: packages/ckeditor5-image/src/imageresize/imageresizeediting.js [hunk 3]
```diff
@@ -45,55 +37,15 @@ export default class ImageResizeEditing extends Plugin {
...
 	}
 
 	/**
 	 * @private
 	 */
 	_registerSchema() {
-		this.editor.model.schema.extend( 'image', {
-			allowAttributes: 'width'
+		this.editor.model.schema.extend( 'image', { allowAttributes: 'width' } );
+		this.editor.model.schema.setAttributeProperties( 'width', {
```
Comment (jodator) [inline packages/ckeditor5-image/src/imageresize/imageresizeediting.js:47]: Was this change wasn't required by the issue? If yes - we need tests. If no we need to handle this in a separate issue to know why this change and when it is needed.

File: packages/ckeditor5-image/tests/imageresize/imageresizehandles.js [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+describe( 'ImageResizeHandles', () => {
+	// 100x50 black png image
+	const IMAGE_SRC_FIXTURE = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAyCAQAAAAAPLY1AAAAQklEQVR42u3PQREAAAgDoK1/' +
+		'aM3g14MGNJMXKiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiJysRFNMgH0RpujAAAAAElFTkSuQmCC';
+
+	let absoluteContainer, widget, editor, view, viewDocument, editorElement;
+
+	before( () => {
+		// This container is required to position editor element in a reliable element.
+		// @todo ensure whether it's required after migrating the tests.
```
Comment (jodator) [inline packages/ckeditor5-image/tests/imageresize/imageresizehandles.js:43]: `todo` ensure whether it's required after migrating the tests.

File: packages/ckeditor5-widget/src/widgetresize.js [hunk 5]
```diff
@@ -87,7 +87,7 @@ export default class WidgetResize extends Plugin {
 			}
 		};
 
-		const redrawFocusedResizerThrottled = throttle( redrawFocusedResizer, 200 ); // 5fps
+		const redrawFocusedResizerThrottled = throttle( redrawFocusedResizer, 50 ); // 5fps
```
Comment (jodator) [inline packages/ckeditor5-widget/src/widgetresize.js:90]: FPS changed:

```suggestion
		const redrawFocusedResizerThrottled = throttle( redrawFocusedResizer, 50 ); // 20fps
```

**Comment by jodator** [comment]
@panr unfrotunately, please remove the config option. It's easier to add it later then removed unused ones.

If something is doable by adding/removing plugins we prefer that over a configuration.

File: packages/ckeditor5-image/src/imageresize/imageresizeediting.js (hunk 3 above)
Comment (panr) [reply packages/ckeditor5-image/src/imageresize/imageresizeediting.js:47]: After splitting the editing plugin I had to add this (it based on our convention to mark certain attributes as `isFormatting`). The line is covered by this test https://github.com/ckeditor/ckeditor5/blob/4fcdefe218d262441ff5a5275dae28f610ce7092/packages/ckeditor5-image/tests/imageresize/imageresizeediting.js#L121.

Comment (jodator) [reply packages/ckeditor5-image/src/imageresize/imageresizeediting.js:47]: I've missed that :( sorry. Let's leave it then.

File: packages/ckeditor5-widget/src/widgetresize.js (hunk 5 above)
Comment (panr) [reply packages/ckeditor5-widget/src/widgetresize.js:90]: How did you measure this?

Comment (jodator) [reply packages/ckeditor5-widget/src/widgetresize.js:90]: I've just updated the value in the comment to match `T = 50ms` - FPS should grow 4 times.

Comment (jodator) [reply packages/ckeditor5-widget/src/widgetresize.js:90]: If the comment itself is rubbish we can remove it.

Comment (panr) [reply packages/ckeditor5-widget/src/widgetresize.js:90]: I'm gonna remove it :D

**Comment by panr** [comment]
I fixed all the minor things. Still don't know how to fix/change the test for `imageresizehandles` to avoid leaks.
